kafka-rest/src/main/java/io/confluent/kafkarest/controllers/TopicManager.java

```diff
@@ -51,13 +52,15 @@
   CompletableFuture<Optional<Topic>> getLocalTopic(String topicName);
 
   /**
-   * Creates a new Kafka {@link Topic}.
+   * Creates a new Kafka {@link Topic} with either partitions count and replication factor
+   * or explicitly specified partition-to-replicas assignments.
    */
   CompletableFuture<Void> createTopic(
       String clusterId,
       String topicName,
       Optional<Integer> partitionsCount,
       Optional<Short> replicationFactor,
+      @Nullable Map<Integer, List<Integer>> replicasAssignments,
```

Comment:
Not nullable. Use empty map for "no replica assignment".

Reply:
Yep, that was one of the things I was considering, but decided to go with a nullable, because that's what the Kafka Admin client does internally (they null-check to determine whether to apply assignments).